spec/support/common_feature_steps.rb

```diff
@@ -28,4 +28,31 @@ def given_I_am_a_2i_reviewer
   def required_permissions_for_2i
     ["signin", "GDS Editor", "2i reviewer", "Unreleased feature"]
   end
+
+  def and_I_am_the_step_by_step_author
+    login_as_user @review_requester
+  end
+
+  def and_I_am_the_reviewer
+    login_as_user @reviewer
+  end
+
+  def then_I_can_see_a_success_message(message)
+    within(".gem-c-success-alert") do
+      expect(page).to have_content message
+    end
+  end
+
+  alias_method :and_I_can_see_a_success_message, :then_I_can_see_a_success_message
+
+  def when_I_click_button(button_text)
+    begin
+      click_button button_text, exact: true
+    rescue Capybara::ElementNotFound
+      # some button-like things are actually marked up as link
+      click_link button_text, exact: true
+    end
+  end
+
+  alias_method :and_when_I_click_button, :when_I_click_button
```

Comment:
I don't think this method now needs to be moved, I'd suggest we should aim to keep it out of common steps as the usage of this is more complex than the capybara equivalents.

For example:

```
  ...
  and_i_click_the_remove_button
end

def and_i_click_the_remove_button
  click_button "Remove"
end
```

mental processing for a reader or debugger is just scroll down the file recognise the capybara DSL, then read up on that if needed.

Whereas with this helper approach:

```
  ...
  and_when_I_click_button "Remove"
  ...
```

mental processing is then: scroll through file looking for method fail to find it so either search code base for it or check each helper file for method. Then discover method is alias so then find actual method. Then process the exception based logic handling and go back to your HTML file to work out which one is in use, and finally question whether exact plays a role.

Reply:
👍 I'll move it back to where it was so that it can be revisited in that file later. You make a good point.